docs/book/src/Overview

```diff
@@ -0,0 +1,20 @@
+# Overview
+
+The reconciliation is like a loop that will run until performing all operations required to ensure a desired state. 
+
+Develop idempotent reconciliation solutions
+When developing operators, the controller’s reconciliation loop needs to be idempotent. By following the [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) you will create [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/) which provide a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster. Breaking this recommendation goes against the design principles of [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) and may lead to unforeseen consequences, such as resources becoming stuck and requiring manual intervention.
+
+Understanding Kubernetes APIs
+Building your operator commonly involves extending the Kubernetes API itself. It is helpful to understand precisely how Custom Resource Definitions interact with the Kubernetes API. Also, the Kubebuilder documentation on Groups and Versions and Kinds may be helpful to understand these concepts better as they relate to operators.
+
+Follow Kubernetes API conventions
+By developing Operator pattern solutions, you will be extending the Kubernetes api. I highly recommend that you understand how Kubernetes API is designed and look for design solutions that follow the same standards and conventions. Please, ensure that you its guidelines
+
+Avoid a design solution where the same controller reconciles more than one Kind
+Having many Kinds (such as CRDs), which are all managed by the same controller, usually goes against the design proposed by controller-runtime. Furthermore, this might hurt concepts such as encapsulation, the Single Responsibility Principle, and Cohesion. Damaging these concepts may cause unexpected side effects and increase the difficulty of extending, reusing, or maintaining the operator.
+
+Use Status Conditionals
+We recommends you to manage your solutions using Status Conditionals. You can check the scaffold provided via the [deploy-image][https://book.kubebuilder.io/plugins/deploy-image-plugin-v1-alpha.html] as an example over how can you implement it.
+
+We recommend checking the documentation on how to [develop Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) solutions from Kubernetes to understand better the purpose of the standard solutions that are built with KubeBuilder.
```

Comment:
```suggestion
## What is "Reconciliation" in Operators?

The reconciliation is a continuos  loop that performs necessary actions on the current state to ensure that the Custom Resource reaches the desired state as specified by the user.

## Why should reconciliations be idempotent?
When developing operators, the controller’s reconciliation loop needs to be idempotent. By following the [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) we create [controllers](https://kubernetes.io/docs/concepts/architecture/controller/) that provide  a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster. Developing idempotent solutions will allow the reconciler to correctly respond to generic or unexpected events, easily deal with application startup or upgrade. More explanation on this is available [here](https://github.com/kubernetes-sigs/controller-runtime/blob/main/FAQ.md#q-how-do-i-have-different-logic-in-my-reconciler-for-different-types-of-events-eg-create-update-delete). 

Writing reconciliation logic according to specific events, breaks the recommendation of operator pattern and goes against the design principles of [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime). This may lead to unforeseen consequences, such as resources becoming stuck and requiring manual intervention.

## Understanding Kubernetes APIs and following API conventions
Building your operator commonly involves extending the Kubernetes API itself. It is helpful to understand precisely how Custom Resource Definitions (CRDs) interact with the Kubernetes API. Also, the [Kubebuilder documentation](https://book.kubebuilder.io/cronjob-tutorial/gvks.html) on Groups and Versions and Kinds may be helpful to understand these concepts better as they relate to operators.

By developing Operator pattern solutions, we extend the Kubernetes api. Understanding the [best practices](https://sdk.operatorframework.io/docs/best-practices/) and following the design patterns is important in the development process. 

## Can my controller reconcile more than one GVK? 

Avoid a design solution where the same controller reconciles more than one Kind. Having many Kinds (such as CRDs), that are all managed by the same controller, usually goes against the design proposed by controller-runtime. Furthermore, this might hurt concepts such as encapsulation, the Single Responsibility Principle, and Cohesion. Damaging these concepts may cause unexpected side effects and increase the difficulty of extending, reusing, or maintaining the operator.

## How to use Status Conditionals
We recommends you to manage your solutions using Status Conditionals. You can check the scaffold provided via the [deploy-image][https://book.kubebuilder.io/plugins/deploy-image-plugin-v1-alpha.html] as an example for its implementation.

Additionally, we recommend checking the documentation on [Operator patterns](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) from Kubernetes to better understand the purpose of the standard solutions that are built with KubeBuilder.
```